src/datasources/email/email.datasource.spec.ts

```diff
@@ -0,0 +1,165 @@
+import { EmailDataSource } from '@/datasources/email/email.datasource';
+import * as postgres from 'postgres';
+import { PostgresError } from 'postgres';
+import { faker } from '@faker-js/faker';
+import { Email } from '@/datasources/email/entities/email.entity';
+import { EmailAddressDoesNotExistError } from '@/datasources/email/errors/email-address-does-not-exist.error';
+import * as shift from 'postgres-shift';
+import configuration from '@/config/entities/__tests__/configuration';
+
+describe('Email Datasource Tests', () => {
+  let target: EmailDataSource;
+  const config = configuration();
+
+  const sql = postgres({
+    host: config.db.postgres.host,
+    port: parseInt(config.db.postgres.port),
+    db: config.db.postgres.database,
+    user: config.db.postgres.username,
+    password: config.db.postgres.password,
+  });
+
+  // Run any pending migration before test execution
+  beforeAll(async () => {
+    await shift({ sql });
+  });
+
+  beforeEach(() => {
+    target = new EmailDataSource(sql);
+  });
+
+  afterEach(async () => {
+    await sql`TRUNCATE emails.signer_emails CASCADE`;
+  });
+
+  afterAll(async () => {
+    await sql.end();
+  });
+
+  it('stores email successfully', async () => {
+    const chainId = faker.number.int({ max: 2147483647 });
```

Comment:
Nitpick: my guess is this number is chosen because it's the maximum positive value for a 32-bit field, but as it is repeated through several tests and the reasoning about it is not super obvious, what do you think about storing it in a variable (that will also provide a name explaining the choice)?

Reply:
Exactly. An integer in Postgres goes up to this value – https://postgresql.org/docs/16/interactive/datatype-numeric.html#DATATYPE-NUMERIC.

Extracted to a `DB_CHAIN_ID_MAX_VALUE` to avoid copying the same value around. [641cb20](https://github.com/safe-global/safe-client-gateway/pull/829/commits/641cb204025e4ca4b2373dd5904ad99710b7da68)